src/poetry/core/factory.py

```diff
@@ -377,6 +407,65 @@ def _configure_package_dependencies(
 
             package.extras = package_extras
 
+    @classmethod
+    def _normalize_depenency_group_names(
```

Comment:
Nit: typo (missing the second `d`; should be `_normalize_dependency_group_names`)